nifi-nar-bundles/nifi-groovyx-bundle/nifi-groovyx-processors/pom.xml

```diff
@@ -38,8 +38,11 @@
         </dependency>
         <dependency>
             <groupId>org.codehaus.groovy</groupId>
-            <artifactId>groovy-all</artifactId>
-            <version>2.4.12</version>
+            <artifactId>groovy-json</artifactId>
+        </dependency>
+        <dependency>
```

Comment:
Should also add groovy-json,datetime,dateutil,xml,templates too just to be safe. I think I saw a new module called groovy-json-direct.

Reply:
`groovy-json` is already a compile-scoped dependency.  Since the current version of `nifi-groovyx-processors` establishes a dependency on `groovy-all`, I'm tempted to list all of groovy's sub-modules in the pom to maintain compatibility with existing flows that use these processors.

The other option is to mention in the migration notes, for the version in which this change is released, that the user should add the groovy submodules needed via the `Additional classpath` property of the `ExecuteGroovyScript` processor configuration.

Reply:
Depending on the module, I think that falls within the scope of our general policy against breaking compatibility on processor functionality abruptly. It's one thing to drop groovy-ant or something like that from groovy-all, but I think we might really inconvenience people who use some of the modules I mentioned.

Reply:
Most likely, all of the modules brought in by `groovy-all` will be included here.  To pare down the default modules, I agree; I think we'll have to wait until NiFi 2.0.

Reply:
I think the best way to handle these dependencies is to list the needed deps necessary to compile the modules in the `nifi-*-processors` POMs as `compile` scope, so that the build and tests will work.  In the NARs' POMs all groovy sub-modules brought in previously by `groovy-all` would be listed so they're available on the classpath via the NarClassLoader.  This will keep the `nifi-*-processors` compile and test phase dependencies minimal, but enable the processor to use the other Groovy sub-modules at runtime.

I'll push an update with these changes by tomorrow.